pact/policy-manager/policy-manager.pact

```diff
@@ -0,0 +1,339 @@
+    , 'guard: (create-capability-guard (ESCROW sale-id))
+    })
+
+  (defun policy-manager-guard:guard ()
+    (create-capability-guard (POLICY_MANAGER))
+  )
+
+  ;; Saves ledger guard information
+  (defschema ledger
+    ledger-guard:guard
+  )
+
+  (deftable ledgers:{ledger}
+    @doc "Designed to save one ledger in a single row")
+
+  (defun enforce-ledger:bool ()
+    @doc "Enforces that function is called from the saved ledger"
+    (with-read ledgers "" {
+      "ledger-guard":= ledger-guard
+      }
+      (enforce-guard ledger-guard)
+    )
+  )
+
+  (defun init:bool(ledger-guard:guard)
+    @doc "Must be initiated with ledger information"
+    (with-capability (GOVERNANCE)
+      (insert ledgers "" {
+        "ledger-guard": ledger-guard
+      })
+    )
+    true
+  )
+
+  ;; Saves Concrete policy information
+  (defschema concrete-policy
+    policy:module{kip.token-policy-v2}
+  )
+
+  (defcap CONCRETE_POLICY:bool (policy-field:string policy:module{kip.token-policy-v2})
+    @doc "Event emission purpose"
+    @event
+    true
+  )
+
+  (deftable concrete-policies:{concrete-policy})
+
+  (defconst NON_FUNGIBLE_POLICY:string 'non-fungible-policy )
+  (defconst ROYALTY_POLICY:string 'royalty-policy )
+  (defconst COLLECTION_POLICY:string 'collection-policy )
+  (defconst GUARD_POLICY:string 'guard-policy )
+  (defconst CONCRETE_POLICY_LIST:[string]
+    [NON_FUNGIBLE_POLICY ROYALTY_POLICY COLLECTION_POLICY GUARD_POLICY] )
+
+  (defun write-concrete-policy:bool (policy-field:string policy:module{kip.token-policy-v2})
+    (contains policy-field CONCRETE_POLICY_LIST)
+    (with-capability (GOVERNANCE)
+      (write concrete-policies policy-field {
+        "policy": policy
+        }
+      )
+      (emit-event (CONCRETE_POLICY policy-field policy))
+    true)
+  )
+
+  (defun get-concrete-policy:module{kip.token-policy-v2} (policy-field:string)
+    (with-read concrete-policies policy-field {
+      "policy":= policy
+      }
+      policy)
+  )
+
+  ;; Capbilities to guard internal functions
+
+  (defcap OFFER:bool
+    ( sale-id:string
+    )
+    @doc "Capability to grant internal transaction inside OFFER"
+    true
+  )
+
+  (defcap BUY:bool
+    ( sale-id:string
+    )
+    @doc "Capability to grant internal transaction inside BUY"
+    true
+  )
+
+  (defcap WITHDRAW:bool
+    ( sale-id:string
+    )
+    @doc "Capability to grant internal transaction inside WITHDRAW"
+    true
+  )
+
+
+  ;; Map list of policy functions
+
+  (defun enforce-init:[bool]
+    (token:object{token-info})
+    (enforce-ledger)
+    (with-capability (POLICY_MANAGER)
+      (map-init token (at 'policies token))
+    )
+  )
+
+  (defun enforce-mint:[bool]
+    ( token:object{token-info}
+      account:string
+      guard:guard
+      amount:decimal
+    )
+    (enforce-ledger)
+    (with-capability (POLICY_MANAGER)
+      (let ((policies:[module{kip.token-policy-v2}]  (at 'policies token)))
```

Comment:
`enforce-init` doesn't need a let, can these do the same? (there is a small cost for `let`)

Reply:
yes